packages/core/database/src/migrations/internal.ts

```diff
@@ -4,26 +4,36 @@ import { wrapTransaction } from './common';
 import { internalMigrations } from './internal-migrations';
 import { createStorage } from './storage';
 
-import type { MigrationProvider } from './common';
+import type { InternalMigrationProvider, Migration } from './common';
 import type { Database } from '..';
 
-export const createInternalMigrationProvider = (db: Database): MigrationProvider => {
+export const createInternalMigrationProvider = (db: Database): InternalMigrationProvider => {
   const context = { db };
+  const migrations: Migration[] = [...internalMigrations];
 
-  const umzugProvider = new Umzug({
-    storage: createStorage({ db, tableName: 'strapi_migrations_internal' }),
-    logger: console,
-    context,
-    migrations: internalMigrations.map((migration) => {
-      return {
-        name: migration.name,
-        up: wrapTransaction(context.db)(migration.up),
-        down: wrapTransaction(context.db)(migration.down),
-      };
-    }),
-  });
+  const createProvider = () => {
+    return new Umzug({
+      storage: createStorage({ db, tableName: 'strapi_migrations_internal' }),
+      logger: console,
+      context,
+      migrations: migrations.map((migration) => {
```

Comment:
I seem to remember we can provide a function to lazy load the migrations here in taht case you would avoid recreating the umzug instance on every register, probably simpler.

Else you can could do sth like that


```ts
const createInternalMigrations = () => {
	const migrations = [...defaults];
	let umzug;
  
  const getUmzug() => {
    if  (umzug) {
      return umzug
    }
    
    umzug =  new Umzug(...)
    return umzug
  }

return {
	register(m: Migration) {
	
		if (umzug !== undefined) {
		  throw new Error('provider already loaded, you need to register during the registration phase')
		}
	
	  validateMigration(m)
		migrations.push(m)
	},
	shouldRun() {
	  getUmzug().shouldRun()
	},
	up() {
	  getUmzug().up()	
	},
	down() {
		  getUmzug().down()
	}
};
}
```